### ISSUE TITLE
Port number not always being passed to remotes correctly

### ISSUE THREAD

**Issue by uspen** [pgbackrest/pgbackrest#424]
Hello! pgbackrest 1.24

Backup host:

```
[stanza]
db1-host=1node0012.db.example.com
db1-port=5432
db1-path=/data/pg/stanza
db1-user=postgres
db2-host=0node0014.db.example.com
db2-port=5432
db2-path=/data/pg/stanza
db2-user=postgres
db3-host=2node0020.db.example.com      <-- master
db3-port=5433                          <-- different port
db3-path=/data/pg/stanza
db3-user=postgres
```
Database host (2node0020.db.example.com):

```
[global]
repo-path=/data/pgbackrest
backup-host=backup.db.example.com
backup-user=postgres
[stanza]
db-path=/data/pg/stanza
db-port=5433

```

LOG:

```
2017-10-11 12:01:41.946 P00  DEBUG:     <SKIP> postgres@2node0020.db.example.com '/usr/bin/pgbackrest --buffer-size=4194304 --command=backup --compress-level=6 --compress-level-network=3 --db1-path=/data/pg/stanza --db1-port=5432 ####  USED FIRST PORT FROM CONFIG #### --protocol-timeout=1830 --repo-path=/data/pgbackrest --stanza=stanza --type=db remote', strId = '2node0020.db.example.com remote', strName = remote
2017-10-11 12:01:42.375 P00  DEBUG:     Db::connect(): bWarnOnError = true
WARN [056]: : raised on '2node0020.db.example.com remote' host: не удалось подключиться к серверу: Нет такого файла или каталога
	Он действительно работает локально и принимает
	соединения через доменный сокет "/var/run/postgresql/.s.PGSQL.5432"?
<SKIP>
ERROR [056]: : unable to find master database - cannot proceed
```

If I change FIRST port to 5433 backup work:

Backup host:

```
[stanza]
db1-host=1node0012.db.example.com
db1-port=5433                          <-- change THIS port from 5432
db1-path=/data/pg/stanza
db1-user=postgres
db2-host=0node0014.db.example.com
db2-port=5432
db2-path=/data/pg/stanza
db2-user=postgres
db3-host=2node0020.db.example.com      <-- master
db3-port=5433                          <-- different port
db3-path=/data/pg/stanza
db3-user=postgres
```

LOG:

```
2017-10-11 12:33:32.923 P00  DEBUG:     <SKIP> strCommand = /usr/bin/pgbackrest --buffer-size=4194304 --command=backup --compress-level=6 --compress-level-network=3 --db1-path=/data/pg/stanza --db1-port=5433 --protocol-timeout=1830 --repo-path=/data/pgbackrest --stanza=stanza --type=db remote, strCommandSSH = ssh, strHost = 2node0020.db.example.com, strUser = postgres
2017-10-11 12:33:32.923 P00  DEBUG:     <SKIP>
2017-10-11 12:33:33.355 P00  DEBUG:     Db::connect(): bWarnOnError = true
2017-10-11 12:33:33.483 P00  DEBUG:     Db::connect=>: bResult = true
```

**Comment by dwsteele** [comment]
Well, I thought we had squished this one.

We'll look at it for the next release.  Meanwhile, if you specify the port in the db config only I believe it will work.

**Comment by dwsteele** [comment]
Fixed at ac54aac448e1e.  Sorry it took so long.  I had to refactor and write new tests (c77fc1fa61582e19) in order to find the issue, even though it ended up being a simple change.

**Comment by uspen** [comment]
Thank you!
